flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/batch/sql/SetOperatorsTest.scala

```diff
@@ -178,4 +182,54 @@ class SetOperatorsTest extends TableTestBase {
       expected
     )
   }
+
+  @Test
+  def testUnionNullableTypes(): Unit = {
```

Comment:
I think this should also be added to stream API as well but I am not sure which test category it falls into.